java/google/registry/flows/FlowMetadata.java

```diff
@@ -0,0 +1,24 @@
+package google.registry.flows;
+
+import com.google.auto.value.AutoValue;
+import google.registry.model.ImmutableObject;
+
+/** Object to allow setting and retrieving metadata information in flows. */
+@AutoValue
+public abstract class FlowMetadata extends ImmutableObject {
+
+  public abstract boolean superuser();
```

Comment:
I think it's slightly less self-evident than the ImmutableList<DomainCheck> personally, and I guess since it's a pretty consequential bit it seems worth the explicitness about stating what it means.  And yeah, javadoc is a bit inconsistent overall but I think especially for things that are going to be "APIs" it's worth erring on the side of more documentation.